File: JavaScript/rotate-string.js
/**
 * @param {string} s
 * @param {string} goal
 * @return {boolean}
 */
var rotateString = function (s, goal) {
    if (s.length != goal.length) return false
    for (let shift = 0; shift < s.length; shift++) {
        let equals = true
        for (let i = 0; i < s.length; i++) {
            const j = (i + shift) % s.length
            if (s[i] !== goal[j]) {
                equals = false
                break
            }
        }
        if (equals) {
            return true
        }
    }
    return false
};

File: Rust/rotate-string.rs
impl Solution {
    pub fn rotate_string(s: String, goal: String) -> bool {
        if s.len() != goal.len() {
            return false;
        }
        let s = s.as_bytes();
        let goal = goal.as_bytes();
        for shift in 0..s.len() {
            let mut equals = true;
            for (i, c) in s.iter().enumerate() {
                let j = (i + shift) % s.len();
                if *c != goal[j] {
                    equals = false;
                    break;
                }
            }
            if equals {
                return true;
            }
        }
        false
    }
}